### PR TITLE
return simple docs to methods

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

the separate docking does not go well with IDE which uses status docstring paring to methods/functions
To keep it simple we shall not include doctests and most of the examples move to pure RST docs

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
@Borda for trainer args, do you think it would help to sort them alphabetically, the doc string I mean.

**Comment by Borda** [comment]
> @Borda for trainer args, do you think it would help to sort them alphabetically, the doc string I mean.

do you think that there is a tool for it or do we have to make it manually?

**Comment by awaelchli** [comment]
I don't know any tool for that. Doing it manually for Trainer one time I think is worth it.

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -279,6 +403,19 @@ def fit(
...
+
+        Args:
+            model: Model to fit.
+
+            train_dataloader: A Pytorch DataLoader with training samples. If the model has
+                a predefined train_dataloader method this will be skipped.
+
+            val_dataloaders: Either a single Pytorch Dataloader or a list of them, specifying validation samples.
+                If the model has a predefined val_dataloaders method this will be skipped
+
```
Comment (SkafteNicki) [inline pytorch_lightning/trainer/trainer.py:417]: `datamodule` seems to missing in the arg description
```suggestion
datamodule: A instance of :class:`LightningDataModule`.
```

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -522,6 +659,24 @@ def test(
...
+            ckpt_path: Either ``best`` or path to the checkpoint you wish to test.
+                If ``None``, use the weights from the last epoch to test. Default to ``best``.
+
+            model: The model to test.
+
+            test_dataloaders: Either a single
+                Pytorch Dataloader or a list of them, specifying validation samples.
+
+            verbose: If True, prints the test results
+
```
Comment (SkafteNicki) [inline pytorch_lightning/trainer/trainer.py:676]: Again here
```suggestion
datamodule: A instance of :class:`LightningDataModule`.
```

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (justusschock) [reply pytorch_lightning/trainer/trainer.py:417]: can we find a more meaningful description for this?

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (justusschock) [reply pytorch_lightning/trainer/trainer.py:676]: can we find a more meaningful description for this?

**Comment by awaelchli** [comment]
Is this good with @williamFalcon?

**Comment by Borda** [comment]
> Is this good with @williamFalcon?

he is in reviewers, and having some extra file with docstring is not a solution for long docstring neither readability... so if we want to make the docs shorter, lets make minimal docstrings and add more info in RST docs

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Pull request merged**